test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+	log.Info("Get 3scale admin Token")
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system-seed",
+		},
+	}
+	err := client.Get(ctx, k8sclient.ObjectKey{Name: s.Name, Namespace: threeScaleNamespace}, s)
+	if err != nil {
+		log.Error("Error get 3Scale admin token ", err)
+		return "", err
+	}
+	accessToken := string(s.Data["ADMIN_ACCESS_TOKEN"])
+	//fmt.Printf(" 🔑  Found 3scale admin token access token: %s\n", accessToken)
+	return accessToken, nil
+}
+
+func getThreeScaleAdminPortal(ctx context.Context, client k8sclient.Client) (string, error) {
+	log.Info("Get 3scale admin Portal")
+	opts := k8sclient.ListOptions{
+		Namespace: threeScaleNamespace,
+	}
+	routes := routev1.RouteList{}
+	err := client.List(ctx, &routes, &opts)
+	if err != nil {
+		log.Info("Error obtaining routes list in " + threeScaleNamespace + " namespace  ")
+		return "", err
+	}
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, "3scale-admin") {
+			log.Info("Route: " + route.Name)
+			annotations_map := route.GetObjectMeta().GetAnnotations()
+			admin_portal := annotations_map["zync.3scale.net/host"]
+			log.Info("3scale Admin portal: " + admin_portal)
+			return admin_portal, nil
+		}
+	}
+	return "", nil
+}
+
+func createAdminPortalCredentialsSecret(ctx context.Context, client k8sclient.Client,
+	token3scale string, admin_portal string) error {
+	log.Info("Create adminportal-credentials secret")
+	admin_portal_url := "https://" + token3scale + "@" + admin_portal
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "adminportal-credentials",
+			Namespace: apicastNamespace,
+		},
+		Data: map[string][]byte{},
+	}
+	secret.Data["AdminPortalURL"] = []byte(admin_portal_url)
+	err := client.Create(ctx, secret)
+	if err != nil {
+		log.Error("Error create adminportal-credentials Secret", err)
+		return err
+	}
+	return nil
+}
+
+// Use self-managed APIcast instead of the builded one for API
+// This is attempt to automate the Promotion steps in 3scale Admin Portal
+// Require more investigation. Recommended to use manual meanwhile
+func deleteManagedApicastRoutes(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Simulate Promotion in 3scale Admin Portal - Delete api-3scale-apicast- routes in " + threeScaleNamespace + " namespace")
+	routes := routev1.RouteList{}
+	err := client.List(context.TODO(), &routes, &k8sclient.ListOptions{
+		Namespace: threeScaleNamespace,
+	})
+	if err != nil {
+		log.Error("Error obtaining routes list in "+threeScaleNamespace+" namespace  ", err)
+		return err
+	}
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, "api-3scale-apicast-") {
+			err := client.Delete(ctx, &route)
+			if err != nil {
+				log.Error("error Delete Route", err)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Install 3scale APIcast gateway operator
+func installThreeScaleApicastGatewayOperator(ctx context.Context, client k8sclient.Client) error {
```

Comment:
```suggestion
func installThreeScaleApicastGatewayOperator(client k8sclient.Client) error {
```
context isn't used in the function. 👍

Reply:
Done